### DISCUSSION TITLE
fooocus-control hanging mid-generation

### DISCUSSION THREAD

**Discussion by prefetchgif** [fenneishi/Fooocus-Control#5] [Q&A]
seems to happen often (but not always.)  not sure why it's hanging.  no resources are being used, and no errors are output.

this is a fresh install (today) and it is using a single input image.

i check the output folder and see images in there.  when i refresh the browser, and type a new prompt (eg. 'cat') it will then appear to generate the _previous_ images and display those, but not a cat.  then i do a hard refresh again and put in 'dog' and it will then appear to generate a cat. 

seems like when it gets hung, and the browser is refreshed, that it lags behind.  

ideas?

<img width="1150" alt="Screen Shot 2023-10-26 at 12 38 08 PM" src="https://github.com/fenneishi/Fooocus-Control/assets/45082312/8ddb011f-7f15-483a-bb62-ac910df7b95f">
<img width="748" alt="Screen Shot 2023-10-26 at 12 37 06 PM" src="https://github.com/fenneishi/Fooocus-Control/assets/45082312/1ce51bec-9d89-46b9-af4d-e62dd936869b">
<img width="650" alt="Screen Shot 2023-10-26 at 12 36 48 PM" src="https://github.com/fenneishi/Fooocus-Control/assets/45082312/2d17af81-cf28-41ef-a4cd-e9c5fc326acd">

**Comment by PeiPeiWow** [comment]
Is it a network problem? Maybe it's because you're using the online version, as models are downloaded as needed in the online version.

**Comment by fenneishi** [comment]
I basically understand what you mean, I don't know what the reason is now, when I will try to reproduce.
It would be helpful if you could provide more log information.
![image](https://github.com/fenneishi/Fooocus-Control/assets/33345538/abf405ff-85aa-4759-b9e9-bc6973c8ac0e)
You can try using fooocus directly to see if you have the same problem

**Reply by prefetchgif** [reply]
okay, i think i figured it out.

the problem is that i had several open browser windows to the fooocus instance and it was confusing it and causing the browser client side to hang.

at some point it would be nice to be able to run multiple windows (eg. in a multi-user or team environment where people are using the same fooocus server) - but for now i'll just make sure i only have a single browser open at a time.

**Reply by prefetchgif** [reply]
i might have spoken too soon.  it looks like the browser GUI hangs a lot less with a single browser open, but it still sometimes hangs, and when it does, it's impossible to get it to re-sync with the server side.  for now i will just restart the server side when it hangs.

i think the solution would be to have some kind of "sync up" button that would do a soft reset on the browser GUI and make it so it display the most recent images instead of displaying the previous images.  not sure if that makes sense.

**Comment by fenneishi** [comment]
Try using fooocus directly to see if you have the same problem and give me the log.html
![image](https://github.com/fenneishi/Fooocus-Control/assets/33345538/75ee01b1-d3c6-48b2-b96c-40e2bd3bc221)

**Comment by prefetchgif** [comment]
looks like it's a fooocus bug.

[(https://github.com/lllyasviel/Fooocus/issues/703)](https://github.com/lllyasviel/Fooocus/issues/703)

**Comment by prefetchgif** [comment]
okay, so this is supposed to be fixed version '2.1.771' of Fooocus, but FooocusControl is on '2.1.701'

what's the best way to keep Fooocus up to date w/ FooocusControl?  is that something we should do manually, or is there a plan to keep the hard fork?

btw, FooocusControl is awesome, thanks for the work you've done on it.

**Reply by fenneishi** [reply]
is there a plan to keep the hard fork?

---

yes

**Reply by prefetchgif** [reply]
okay, cool - so are you going to fold in bug fixes from fooocus?
